### PR TITLE
Fix some diagnostics bugs

### PR DESCRIPTION
**Pull request by kpamnany**
# Description

A bug in 'DumpStateAndAux'. Also clean up some issues caused by the concurrent merge of 1019 and 961. Finally, enable diagnostics collection in two experiments on Slurm CI, to ensure all diagnostics groups run at least once on CI.

I have

- [X] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/format.jl`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [X] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by kpamnany** [comment]
bors try

**Comment by akshaysridhar** [comment]
Tested offline with the GCM configuration - output for one of the `write` steps
```
┌ Info: Update
│     simtime = 2204288.75 / 129600000.00
│     runtime = 00:04:59
└     norm(Q) = 8.0614480936960000e+13
┌ Info: Diagnostics: DumpStateAndAux
└     collecting at 2.5920815e6
```

File: slurmci-test.toml [hunk 1]
```diff
@@ -49,11 +49,11 @@ gpu = [
   { file = "test/Numerics/DGmethods/advection_diffusion/pseudo1D_heat_eqn.jl", slurmargs = ["--ntasks=3"], args = ["--integration-testing"] },
   { file = "test/Numerics/DGmethods/advection_diffusion/periodic_3D_hyperdiffusion.jl", slurmargs = ["--ntasks=3"], args = ["--integration-testing"] },
   { file = "tutorials/Atmos/dry_rayleigh_benard.jl", slurmargs = ["--ntasks=3", "--time=01:30:00"], args = [] },
-  { file = "experiments/AtmosGCM/heldsuarez.jl", slurmargs = ["--ntasks=3", "--time=01:30:00"], args = [] },
+  { file = "experiments/AtmosGCM/heldsuarez.jl", slurmargs = ["--ntasks=3", "--time=01:30:00"], args = ["--diagnostics default"] },
```
Comment (simonbyrne) [inline slurmci-test.toml:52]:
```suggestion
  { file = "experiments/AtmosGCM/heldsuarez.jl", slurmargs = ["--ntasks=3", "--time=01:30:00"], args = ["--diagnostics=default"] },
```

File: slurmci-test.toml [hunk 2]
```diff
@@ -49,11 +49,11 @@ gpu = [
...
   { file = "experiments/AtmosLES/risingbubble.jl", slurmargs = ["--ntasks=3"], args = [] },
   { file = "experiments/AtmosLES/surfacebubble.jl", slurmargs = ["--ntasks=3", "--time=01:30:00"], args = [] },
   { file = "experiments/AtmosLES/dycoms.jl", slurmargs = ["--ntasks=3"], args = [] },
-  { file = "experiments/AtmosLES/bomex.jl", slurmargs = ["--ntasks=3"], args = [] },
+  { file = "experiments/AtmosLES/bomex.jl", slurmargs = ["--ntasks=3"], args = ["--diagnostics default"] },
```
Comment (simonbyrne) [inline slurmci-test.toml:56]:
```suggestion
  { file = "experiments/AtmosLES/bomex.jl", slurmargs = ["--ntasks=3"], args = ["--diagnostics=default"] },
```

**Pull request merged**